docs/DeveloperGuide.adoc

```diff
@@ -452,27 +452,176 @@ with current implementations (able to wrap `ObservableList` and sort the content
+
+// tag::processingMarkDoneDG[]
+
+=== ProcessingMarkDone feature
+This section describes how the processingMarkDone feature is implemented.
+
+ProcessingMarkDone is facilitated by the `ProcessingMarkDoneCommand` class, which extends the `Command` class.
+
+ProcessingMarkDone is done by accessing `Patient` and `Donor` in `ModelManager`,
+then, it will get the `ProcessingList` from the `Donor` and then marking a task as done.
+
+It implements the `ProcessingMarkDoneCommand#isValidDonorPatientPair(Nric firstNric, Nric secondNric, Model model)` boolean operation. +
+This method works similar to `ProcessingCommand#isValidDonorPatientPair(Nric firstNric, Nric secondNric, Model model)` method but with a slight difference. +
+This method just need to check the `Status` of the `Donor` and `Patient` object and make sure it is `processing`. +
+However, this method still need to determine which Nric belongs to a patient and which Nric belongs to a donor
+since the input allows user to give two Nric in no particular order. +
+A code snippet of the method is shown below: This part of the method is different from the `ProcessingCommand#isValidDonorPatientPair(Nric firstNric, Nric secondNric, Model model)` method.
+
+```
+if (model.hasPatient(patientNric) && model.hasDonor(donorNric)
+        && patient.getStatus().isProcessing() && donor.getStatus().isProcessing()
+        && match(donor, patient)) {
+    return true;
+} else {
+    return false;
+}
+```
+
+When the user execute the processingMarkDone feature, the method, `ProcessingMarkDoneCommand#isValidDonorPatientPair(Nric firstNric, Nric secondNric, Model model)` will be called
+to determine if the user give a valid patient and donor input. +
+It will also check if the task number given is valid. +
+Then, it will call `Donor#markTaskAsDone(int taskNumber)` to mark the task on the `ProcessingList` of that donor.
+It will then display the list out.
+
+Following is an object diagram to show the processingMarkDone feature
+
+TODO diagram
+
+// end::processingMarkDoneDG[]
+
+// tag::doneDG[]
+
+=== Done feature
+This section describes how the done feature is implemented.
+
+Done is facilitated by the `DoneCommand` class, which extends the `Command` class.
+
+Done is done by accessing `Patient` and `Donor` in `ModelManager`,
+then determine if the patient and donor can match each other or not.
+
+It implements the `DoneCommand#isPass(String result)` boolean operation. +
+This method check if the result of the cross-matching test. +
+
+If the patient and donor did a cross-matching test and the result of the test is a fail,
+then the patient and donor will need to be send back to find new matches and they
+should not be able to match each other anymore. +
+`Donor` will have an ArrayList `patientsMatchedBefore` which stores all the patients' Nric which fails to match with it.
+Then, the patient and donor will have their `Status` set to `not processing` so that it will be able to find a match in ORGANice.
```

Comment:
So that they can find another match and be considered in future matching results.

Reply:
done